feature/user_profile/src/main/java/org/mifos/mobile/feature/user_profile/screens/UserProfileScreen.kt

```diff
@@ -0,0 +1,324 @@
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.tooling.preview.PreviewParameterProvider
+import androidx.compose.ui.unit.dp
+import androidx.core.content.ContextCompat
+import androidx.core.graphics.drawable.toBitmap
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import org.mifos.mobile.core.common.Network
+import org.mifos.mobile.core.common.utils.DateHelper
+import org.mifos.mobile.core.datastore.PreferencesHelper
+import org.mifos.mobile.core.model.entity.client.Client
+import org.mifos.mobile.core.model.entity.client.Group
+import org.mifos.mobile.core.ui.component.MFScaffold
+import org.mifos.mobile.core.ui.component.MifosProgressIndicatorOverlay
+import org.mifos.mobile.core.ui.component.MifosUserImage
+import org.mifos.mobile.core.ui.component.NoInternet
+import org.mifos.mobile.core.ui.component.UserProfileField
+import org.mifos.mobile.core.ui.component.UserProfileTopBar
+import org.mifos.mobile.feature.third.party.user_profile.R
+import org.mifos.mobile.feature.user_profile.utils.TextDrawable
+import org.mifos.mobile.feature.user_profile.utils.UserDetailState
+import org.mifos.mobile.feature.user_profile.utils.UserDetails
+import org.mifos.mobile.feature.user_profile.viewmodel.UserDetailViewModel
+
+/**
+ * @author pratyush
+ * @since 20/12/2023
+ */
+
+@Composable
+fun UserProfileScreen(
+    viewModel: UserDetailViewModel = hiltViewModel(),
+    navigateBack: () -> Unit,
+    changePassword: () -> Unit
+) {
+
+    val context = LocalContext.current
+    val uiState by viewModel.userDetailUiState.collectAsStateWithLifecycle()
+    var isOnline by remember { mutableStateOf(false) }
+
+    isOnline = Network.isConnected(context)
+
+    UserProfileScreen(
+        navigateBack = navigateBack,
+        uiState = uiState,
+        changePassword = changePassword,
+        isOnline = isOnline
+    )
+}
+
+@Composable
+fun UserProfileScreen(
+    navigateBack: () -> Unit,
+    uiState: UserDetailState,
+    changePassword: () -> Unit,
+    isOnline: Boolean
+) {
+    val context = LocalContext.current
+    val snackBarHostState = remember {
+        SnackbarHostState()
+    }
+    var userBitmap : Bitmap? = getUserImage( null, context)!!
+    var client : Client = Client()
+
+    MFScaffold(
+        topBarTitleResId = R.string.user_details,
+        navigateBack = navigateBack,
+        snackbarHost = { SnackbarHost(hostState = snackBarHostState) },
+        scaffoldContent = { paddingValues ->
+
+            Box(
+                Modifier
+                    .fillMaxSize()
+            ) {
+                Column(
+                    modifier = Modifier
+                        .fillMaxSize(),
+                ) {
+
+                    when (uiState) {
+                        is UserDetailState.ShowError -> {
+                            LaunchedEffect(true) {
+                                snackBarHostState.showSnackbar(
```

Comment:
call `MifosErrorComponent` here
message = uiState.message
onRetry = //perform reload

check for network connectivity here ....  `Onclick`